src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.WebHost.Management;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Azure.WebJobs.Script.WebHost.Security.Authorization.Policies;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Controllers
+{
+    public class InstanceController : Controller
+    {
+        private readonly WebScriptHostManager _scriptHostManager;
+        private readonly ScriptSettingsManager _settingsManager;
+        private readonly IInstanceManager _instanceManager;
+
+        public InstanceController(WebScriptHostManager scriptHostManager, ScriptSettingsManager settingsManager, IInstanceManager instanceManager)
+        {
+            _scriptHostManager = scriptHostManager;
+            _settingsManager = settingsManager;
+            _instanceManager = instanceManager;
+        }
+
+        [HttpPost]
+        [Route("admin/instance/assign")]
+        //[Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public IActionResult Assign([FromBody] EncryptedAssignmentContext encryptedAssignmentContext)
```

Comment:
Why the new controller for this - aren't these just host level operations that belong in the existing HostController?

Reply:
Main reason is I didn't wanna pollute the `/admin/host` path with these APIs. There is some sort of overlap between instance status/info and host status/info. The assign doesn't overlap, but it's up to you. Would you rather move the assign call to the HostController?